### ISSUE TITLE
Legion: Regions considered escaping from non-toplevel replicated tasks even if destroyed

### ISSUE THREAD

**Issue by manopapad** [StanfordLegion/legion#960]
The attached testcase [escape.zip](https://github.com/StanfordLegion/legion/files/5462531/escape.zip), when compiled against latest control_replication on sapling and invoked with:

```
mpirun -H n0000,n0001 --bind-to none ./escape -level mapper=2
```

produces the error:

```
[0 - 7fb9e9c0c800]    0.047134 {6}{runtime}: [fatal 2009] LEGION FATAL: Returnable privileges are not currently supported for control replicated tasks that are are not the top-level task such as task task_foo (UID 8) (from file /home/manolis/legion/runtime/legion/legion_context.cc:11109)
For more information see:
http://legion.stanford.edu/messages/fatal_code.html#fatal_code_2009
```

CC @gshipman

**Comment by gshipman** [comment]
@manopapad @lightsighter , I refactored my code to only replicate the TLT.  I'm still seeing the error about Logical regions escaping non-top-level replicated task..   
```Region LogicalRegion(1,IndexSpace(1,1),FieldSpace(1)) is escaping from non-top-level replicated task
Region LogicalRegion(1,IndexSpace(1,1),FieldSpace(1)) is escaping from non-top-level replicated task```

**Comment by gshipman** [comment]
@manopapad, @lightsighter After chasing down an unrelated bug I can confirm that I do not end up with the LEGION FATAL above after refactoring to only replicate the TLT and doing all index launches from the TLT. I'm now running in parallel across multiple nodes and GPUs.  I think I can use this as a workaround for a bit.

**Comment by lightsighter** [comment]
If you run this against the `neweq` branch can confirm that it works (it does for me)? If so, I'm likely not to fix it since I'm expecting that `neweq` will merge into `control_replication` sometime in the next month.

**Comment by manopapad** [comment]
The example works on `neweq` for me as well.

**Comment by lightsighter** [comment]
Ok, I'm going to recommend that you use the `neweq` branch for now and close this issue. If there's a reason we need to fix it immediately in the `control_replication` branch then you can reopen it.
